### DISCUSSION TITLE
NPM install error

### DISCUSSION THREAD

**Discussion by jproz** [tteck/Proxmox#2188] [Help Needed]
Gets to this point and fails. 

Installing Node.js...+ bash /dev/fd/63
++ curl -fsSL https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.3/install.sh
=> Downloading nvm from git to '/root/.nvm'
=> Cloning into '/root/.nvm'...
remote: Enumerating objects: 366, done.
remote: Counting objects: 100% (366/366), done.
remote: Compressing objects: 100% (315/315), done.
remote: Total 366 (delta 43), reused 163 (delta 25), pack-reused 0
Receiving objects: 100% (366/366), 365.79 KiB | 2.09 MiB/s, done.
Resolving deltas: 100% (43/43), done.
* (HEAD detached at FETCH_HEAD)
  master
=> Compressing and cleaning up git repository

=> Appending nvm source string to /root/.profile
=> bash_completion source string already in /root/.profile
=> Close and reopen your terminal to start using nvm or run the following to use it now:

export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+ . /root/.bashrc
+ nvm install 16.20.1
bash: line 59: nvm: command not found
++ error_handler 59 '$STD nvm install 16.20.1'
++ local exit_code=127
++ local line_number=59
++ local 'command=$STD nvm install 16.20.1'
++ local 'error_message=\033[01;31m[ERROR]\033[m in line \033[01;31m59\033[m: exit code \033[01;31m127\033[m: while executing command \033[33m$STD nvm install 16.20.1\033[m'
++ echo -e '\n\033[01;31m[ERROR]\033[m in line \033[01;31m59\033[m: exit code \033[01;31m127\033[m: while executing command \033[33m$STD nvm install 16.20.1\033[m'

[ERROR] in line 59: exit code 127: while executing command $STD nvm install 16.20.1
++ [[ 59 -eq 24 ]]
+ exit

Not sure where to proceed.

**Comment by tteck** [comment]
Are you using the Proxmox shell or SSH?

**Reply by tteck** [reply]
The script runs without any problems from this end. 

```
    _   __      _               ____                           __  ___                                 
   / | / /___ _(_)___  _  __   / __ \_________ __  ____  __   /  |/  /___ _____  ____ _____ ____  _____
  /  |/ / __  / / __ \| |/_/  / /_/ / ___/ __ \| |/_/ / / /  / /|_/ / __  / __ \/ __  / __  / _ \/ ___/
 / /|  / /_/ / / / / />  <   / ____/ /  / /_/ />  </ /_/ /  / /  / / /_/ / / / / /_/ / /_/ /  __/ /    
/_/ |_/\__, /_/_/ /_/_/|_|  /_/   /_/   \____/_/|_|\__, /  /_/  /_/\__,_/_/ /_/\__,_/\__, /\___/_/     
      /____/                                      /____/                            /____/             
 
Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 1
Using Root Password: Automatic Login
Container ID: 103
Using Hostname: nginxproxymanager
Using Disk Size: 4
Allocated Cores: 1
Allocated RAM: 1024
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a Nginx Proxy Manager LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using local-lvm for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 103 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.5 
 ✓ Internet Connected
 ✓ DNS Resolved github.com to 140.82.113.4
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Installed Python Dependencies
 ✓ Installed Openresty
 ✓ Installed Node.js
 ✓ Installed Yarn
 ✓ Downloaded Nginx Proxy Manager v2.10.4
 ✓ Set up Enviroment
 ✓ Built Frontend
 ✓ Initialized Backend
 ✓ Created Service
 ✓ Customized Container
 ✓ Started Services
 ✓ Cleaned
 ✓ Completed Successfully!

Nginx Proxy Manager should be reachable by going to the following URL.
         http://192.168.86.5:81 
```

**Reply by jproz** [reply]
I'll try directly from web interface console when I get to my desk.

This was ssh as an admin, 'su -' and then the script. I don't recall having issues with that method before even though I use it rarely (when I'm not at my desk and on my phone). 

(Edit: thanks for the super quick response :) )

**Reply by tteck** [reply]
Well, it does mention "It's suggested to use the Proxmox shell instead of SSH, since SSH can create issues while gathering variables."

**Reply by jproz** [reply]
Yep. I'm 100% aware of that.
In my ignorance, I thought if the script got that far, it was OK but I will try again and report back once I get back to my desk, from the proper console.

**Comment by jproz** [comment]
OK @tteck -- at the console. same errors. both if I just do "default" as well as my custom changes for advanced.
Not sure what else to tell ya. Anything I can get you to troubleshoot? It's not urgent; I can probably just put nginx on the vs a separate LXC.

**Reply by tteck** [reply]
Try clearing browser cache

**Reply by jproz** [reply]
First time I did it in Chrome. 
 
 This time I did it via Safari with a completely cleaned-out cache.
 
 Same error. I wouldn't be surprised if it's something weird on my end; I just don't know where to look.
 I've had *no* issues using your scripts otherwise - running about 10 of them, many of them newly installed as a rotated the services out of docker.
 
 
 ✓ Installed Openresty
+ msg_info 'Installing Node.js'
+ local 'msg=Installing Node.js'
+ echo -ne ' - \033[33mInstalling Node.js...'
 - Installing Node.js...+ bash /dev/fd/63
++ curl -fsSL https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.3/install.sh
=> Downloading nvm from git to '/root/.nvm'
=> Cloning into '/root/.nvm'...
remote: Enumerating objects: 366, done.
remote: Counting objects: 100% (366/366), done.
remote: Compressing objects: 100% (315/315), done.
remote: Total 366 (delta 43), reused 163 (delta 25), pack-reused 0
Receiving objects: 100% (366/366), 365.79 KiB | 2.61 MiB/s, done.
Resolving deltas: 100% (43/43), done.
* (HEAD detached at FETCH_HEAD)
  master
=> Compressing and cleaning up git repository

=> Appending nvm source string to /root/.profile
=> bash_completion source string already in /root/.profile
=> Close and reopen your terminal to start using nvm or run the following to use it now:

export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+ . /root/.bashrc
+ nvm install 16.20.1
bash: line 59: nvm: command not found
++ error_handler 59 '$STD nvm install 16.20.1'
++ local exit_code=127
++ local line_number=59
++ local 'command=$STD nvm install 16.20.1'
++ local 'error_message=\033[01;31m[ERROR]\033[m in line \033[01;31m59\033[m: exit code \033[01;31m127\033[m: while executing command \033[33m$STD nvm install 16.20.1\033[m'
++ echo -e '\n\033[01;31m[ERROR]\033[m in line \033[01;31m59\033[m: exit code \033[01;31m127\033[m: while executing command \033[33m$STD nvm install 16.20.1\033[m'

[ERROR] in line 59: exit code 127: while executing command $STD nvm install 16.20.1
++ [[ 59 -eq 24 ]]
+ exit

**Reply by jproz** [reply]
@tteck ya ... at a loss. Shrug .. I'll come up with another way I guess since you closed this. 
Very much appreciate your work!

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/issues/2200

**Reply by jproz** [reply]
I'm doing a little dance - you nailed it. The combination of the change you made + change login shell to bash + clear cache seems to be the ticket. Thanks @tteck
